common/main/java/com/couchbase/lite/internal/replicator/AbstractCBLWebSocket.java

```diff
@@ -364,12 +376,46 @@ private OkHttpClient setupOkHttpClient() throws GeneralSecurityException {
         final Authenticator authenticator = setupBasicAuthenticator();
         if (authenticator != null) { builder.authenticator(authenticator); }
 
+        // cookies
+        setupCookieJar(builder);
```

Comment:
What would you think of renaming the method getCookieJar, and having it just return the anonymous class?  Then you wouldn't have to pass the builder.  Just a thought...

Reply:
I tried to follow the existing code that passes the builder. I could update as you suggested as well.